basicdataset/src/main/java/ai/djl/basicdataset/tabular/CsvTranslator.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.basicdataset.tabular;
+
+import ai.djl.ndarray.NDArray;
+import ai.djl.ndarray.NDList;
+import ai.djl.translate.TranslateException;
+import ai.djl.translate.Translator;
+import ai.djl.translate.TranslatorContext;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.csv.CSVRecord;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/** A {@link Translator} that converts between CSV text and {@link NDList}. */
+public class CsvTranslator implements Translator<String, String> {
+
+    private final CSVFormat csvFormat;
+
+    /**
+     * Constructs a CsvTranslator.
+     *
+     * @param arguments the arguments (unused but required for reflection)
+     */
+    @SuppressWarnings({"PMD.UnusedFormalParameter", "deprecation"})
+    public CsvTranslator(Map<String, ?> arguments) {
+        this.csvFormat = CSVFormat.newFormat(',').withRecordSeparator("\n");
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public NDList processInput(TranslatorContext ctx, String csvData) throws TranslateException {
+        try (CSVParser parser = csvFormat.parse(new StringReader(csvData))) {
+            List<float[]> rows = new ArrayList<>();
+            int expectedCols = -1;
+            boolean headerSkipped = false;
+
+            for (CSVRecord record : parser) {
+                if (!headerSkipped && isHeaderRow(record)) {
+                    headerSkipped = true;
+                    continue;
+                }
+
+                if (expectedCols == -1) {
+                    expectedCols = record.size();
+                } else if (record.size() != expectedCols) {
+                    throw new TranslateException(
+                            String.format(
```

Comment:
String.format is unnecessary and slow. just use:

"Row " + record.getRecordNumber() + " has " + record.size() + " columns, expected " + expectedCols